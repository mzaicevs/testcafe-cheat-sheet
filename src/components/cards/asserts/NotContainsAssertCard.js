import React from "react";
import { assert_notContains } from "../../../snippets";
import CodeCard from "../../../base_components/CodeCard";

const NotContainsAssertCard = () => (
  <CodeCard
    cardTitle={assert_notContains.title}
    fullDocLink={assert_notContains.fullDocLink}
    codeSnippet={{
      code: assert_notContains.code,
      language: assert_notContains.language,
      options: assert_notContains.options,
      example: assert_notContains.example
    }}
    description={assert_notContains.description}
  />
);

export default NotContainsAssertCard;
