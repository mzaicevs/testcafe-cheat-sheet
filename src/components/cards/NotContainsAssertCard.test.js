import React from "react";
import { shallow } from "enzyme";
import NotContainsAssertCard from "./NotContainsAssertCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("NotContainsAssertCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<NotContainsAssertCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
