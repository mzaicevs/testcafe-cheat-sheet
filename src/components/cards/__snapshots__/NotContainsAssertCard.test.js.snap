// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotContainsAssertCard matches snapshot 1`] = `
<CodeCard
  cardTitle="Not Contains - Assert"
  codeSnippet={
    Object {
      "code": Array [
        "t.expect( actual ).notContains( unexpected, message, options );",
      ],
      "example": "import { ClientFunction } from 'testcafe';

  fixture \`My fixture\`;
  
  test('My test', async t => {
    const getLocation = ClientFunction(() => document.location.href.toString());

    await t
        .expect(getLocation()).notContains('google.com')
        .expect('foo bar').notContains('baz', 'string does not contain a substring')
        .expect([1, 2, 3]).notContains(4, 'array does not contain a value')
        .expect({ foo: 'bar', hello: 'universe' }).notContains({ buzz: 'abc' }, 'object does not contain a property');
  });",
      "language": "javascript",
      "options": "{ 
    timeout: Number,
      allowUnawaitedPromise: Boolean 
  }",
    }
  }
  description="Asserts that actual does not contain expected."
  fullDocLink="https://devexpress.github.io/testcafe/documentation/test-api/assertions/assertion-api.html#not-contains"
/>
`;
